pep-0458.txt

```diff
@@ -436,53 +468,60 @@ install or upgrade a PyPI project via TUF.
 
 __ https://github.com/theupdateframework/tuf/blob/v0.11.1/docs/TUTORIAL.md#delegate-to-hashed-bins
 
-Based on our findings as of the time of writing, PyPI SHOULD split all targets
-in the *bins* role by delegating them to 1024 delegated roles, each of which
-would sign for PyPI targets whose hashes fall into that "bin" or delegated role
-(see Figure 2).  It was found that 1024 bins would result in the *bins*
-metadata, and each of its delegated roles, being about the same size (40-50KB)
-for about 220K PyPI targets (simple indices and distributions).
+Based on our findings as of the time this document was updated for implementation
+(Oct 7 2019), PyPI SHOULD split all targets in the *bins* role by delegating
+them to 16,384 *bin-n* roles. Each *bin-n* role would sign for the PyPI targets whose
+hashes fall into that bin (see Figure 2).  It was found__
+that this number of bins would result in a 12-17% metadata overhead
+(relative to the average size of downloaded packages) for returning users
+(assuming 256-byte target filenames for all packages), and a 148% overhead
+for new users who are installing pip for the first time.
+
+__ https://docs.google.com/spreadsheets/d/11_XkeHrf4GdhMYVqpYWsug6JNz5ZK6HvvmDZX0__K2I/edit?usp=sharing
+
+This number of bins SHOULD increase when the metadata overhead for returning
+users exceeds 50%. Presently, this SHOULD happen when the number of targets
+increase at least 4x from over 2M to nearly 9M, at which point the metadata
+overhead for returning and new users would be around 49-54% (assuming 256-byte
+target filenames for all packages) and 185% respectively, assuming that the
+number of bins stay fixed. If the number of bins is increased, then the cost
+for all users would effectively be the cost for new users, because their cost
+would be dominated by the (once-in-a-while) cost of downloading the large
+number of delegations in the `bins` metadata. If the cost for new users
```

Comment:
What is the criteria that we would use to determine that the cost for new users is "too much"? Is it the initial download size of the metadata?